### PR TITLE
Set: node version.

### LINKED ISSUE #399: Fairy | Establish consistent strict `node@v20` and `npm@v10`.
### Motivation
1. Well... Versions should be consistent and strict, especially stuff like this!
2. Recently `node@v22` was released, so from as per the conservative approach, one even version before that is due.
3. Some of our GHAs are not set to a strict node/npm version, and we'll cover that here too.

### TODO
1. Add `.nvmrc` file with `v20` (and an EOL 🤓).
2. Add `"engines"` to `package.json` above `"scripts"`:
    ```json
    "engines": {
      "node": "^20",
      "npm": "^10"
    },
    ```
3. Add `.npmrc` file:
    ```ini
    engine-strict=true
    update-notifier=false
    ```
4. Upgrade any `actions/setup-node` to latest version.
5. Request `node@v20` from `actions/setup-node`.
6. Go over all GHAs that use `npm` or `node`, and if they don't use `actions/setup-node`, add it.
7. Add to `package.json` `"scripts"` as the first two:
    ```json
    "scripts": {
      "x": "npm --offline x --",
      "reinstall": "rm -rf node_modules package-lock.json && npm i",
    }
    ```
8. Run `node -v` and make sure you got `v20`.
9. Run `npm -v` and make sure you got `v10`.
10. Run `npm run reinstall` to refresh all dependencies and lock-file.
11. Run `npm i --save-dev @types/node@v20`
12. Add to `.github/dependabot.yml` (the same way it is in `gamitee-client`):
    ```yml
    ignore:
      - dependency-name: "@types/node"
        update-types: ["version-update:semver-major"]

    ```

### Caveats
1. I recommend using `nvm` locally. `nvm` for Windows is also great. If you use it, and something doesn't work, let's adjust.
2. I recommend working from `git-bash` (or something similar, if you know what you're doing), so stuff like `rm -rf` will work.

### PR DESCRIPTION
**Pull request by Pruthvi-iamops**
Closes #399

### CODE REVIEW

File: .github/workflows/deploy.yml [hunk 1]
```diff
@@ -16,6 +16,10 @@ jobs:
       - uses: actions/checkout@v3
         with:
           token: ${{ secrets.GAMITEE_BOT_COMMIT_ACCESS_TOKEN }}
+      - name: Setup Node.js
+        uses: actions/setup-node@v4
+        with:
+          node-version-file: package.json
```
Comment (tomerle) [inline .github/workflows/deploy.yml]: `node-version-file` 🤩

Comment (tomerle) [reply .github/workflows/deploy.yml]: I see logic in calling all setup Actions first, but seems more security-oriented to cal `actions/setup-node` below, just before we actually use `npm` (i.e. `node`).

File: .github/workflows/linter.yml [hunk 2]
```diff
@@ -36,5 +41,4 @@ jobs:
         run: npm run test:coverage
 
       - name: Run ESLint
-        run: npm run eslint:test
-
+        run: npm run eslint:test
```
Comment (tomerle) [inline .github/workflows/linter.yml]: We actually like an EOL (End Of Line) at the EOF (End Of File). 🙏 🤓

File: .npmrc [hunk 3]
```diff
@@ -0,0 +1,2 @@
+engine-strict=true
+update-notifier=false
```
Comment (tomerle) [inline .npmrc]: We actually like an EOL (End Of Line) at the EOF (End Of File). 🙏 🤓

File: package-lock.json
Comment (tomerle) [inline package-lock.json:1]: @Pruthvi-iamops, this is the result of running `npm run reinstall`, right?

**Review by tomerle — CHANGES_REQUESTED** [review]
I only commented on what we see here. Still need to make sure everything requested in the Issue is covered by the PR.

_@Pruthvi-iamops, good stuff in the places you chose solutions better than what I requested._

File: package-lock.json
Comment (Pruthvi-iamops) [reply package-lock.json:1]: @tomerle Yes this are the result of the npm run reinstall which I ran from my local machine.

Comment (tomerle) [reply package-lock.json:1]: Thanks! Asking you was my way of reviewing this. 😎

File: package.json
Comment (tomerle) [inline package.json:1]: @Pruthvi-iamops, please run `npm i --save-dev @types/node@20 && npm ci` and push again. 🙏

**Review by tomerle — APPROVED** [review]

**Pull request merged**